agent/internal/options/options.go

```diff
@@ -81,9 +81,8 @@ type Options struct {
 	// master config.
 	AgentReconnectBackoff int `json:"agent_reconnect_backoff"`
 
-	ContainerRuntime   string             `json:"container_runtime"`
-	SingularityOptions SingularityOptions `json:"singularity_options"`
-	PodmanOptions      PodmanOptions      `json:"podman_options"`
+	ContainerRuntime string        `json:"container_runtime"`
+	PodmanOptions    PodmanOptions `json:"podman_options"`
```

Comment:
We don't want users to be using these configuration options. I'm ok with removing them in the next set of work though if that's what you prefer (I saw one of your comments).

Reply:
I was thinking of having `contianer_runtime `  configuration as is, but will try `PodmanOption` in later iterations.

Reply:
I see Nick likes the `ContainerRuntime` interface and wants to keep it. I think that makes a lot of sense.

I'm saying, we don't need to keep the user-facing `container_runtime` configuration option. Either way, yes we should definitely remove `PodmanOptions` in the next PR if it's not done in this one.

Reply:
I see what you mean here. That's possible and makes sense.